src/mavlink.c

```diff
@@ -124,10 +140,16 @@ void* __MAVLINK_THREAD__(void* arg) {
     for (int i = 0; i < ret; ++i) {
       if (mavlink_parse_char(MAVLINK_COMM_0, buffer[i], &message, &status) == 1) {
         switch (message.msgid) {
-          case MAVLINK_MSG_ID_HEARTBEAT:
-            // handle_heartbeat(&message);
-            break;
-		
+            case MAVLINK_MSG_ID_HEARTBEAT: {
+                mavlink_heartbeat_t heartbeat;
+                mavlink_msg_heartbeat_decode(&message, &heartbeat);
+                if (heartbeat.base_mode & MAV_MODE_FLAG_SAFETY_ARMED) {
+                    dvr_start_recording(dvr);
```

Comment:
as far as I understand, the heartbeat message is sent periodically (eg, once per-second or like that). It looks like `dvr_start_recording` will be called each time we receive the heartbeat message?
Your current solution works because we check if DVR is already started in the DVR thread, but it causes a lot of extra work for DVR thread.
I think better solution would be to have a "current_arm_state" variable defined outside of the loop and only call `start/stop` functions if `current_arm_state` changes?